res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,193 @@
+
+class Component {
+    constructor(group, inKey) {
+        this.group = group;
+        this.inKey = inKey;
+    }
+}
+
+class Deck {
+    constructor(decks) {
+        if (typeof decks === "number") {
+            this.group = Deck.groupForNumber(decks);
+        } else if (Array.isArray(decks)) {
+            this.decks = decks;
+            this.currentDeckNumber = decks[0];
+            this.group = Deck.groupForNumber(decks[0]);
+        }
+    }
+    toggleDeck() {
+        if (this.decks === undefined) {
+            throw Error("toggleDeck can only be used with Decks constructed with an Array of deck numbers, for example [1, 3]");
+        }
+
+        const currentDeckIndex = this.decks.indexOf(this.currentDeckNumber);
+        let newDeckIndex = currentDeckIndex + 1;
+        if (currentDeckIndex >= this.decks.length) {
+            newDeckIndex = 0;
+        }
+
+        this.switchDeck(Deck.groupForNumber(this.decks[newDeckIndex]));
+    }
+    switchDeck(newGroup) {
+        for (let property in this) {
+            if (Object.prototype.hasOwnProperty.call(this, property)) {
+                if (this[property] instanceof Component) {
+                    if (this[property].disconnect !== undefined && typeof this[property].disconnect === "function") {
+                        this[property].disconnect();
+                    }
+                    this[property].group = newGroup;
+                    if (this[property].connect !== undefined && typeof this[property].connect === "function") {
+                        this[property].connect();
+                    }
+                }
+            }
+        }
+    }
+    static groupForNumber(deckNumber) {
+        return "[Channel" + deckNumber + "]";
+    }
+}
+
+class Button extends Component {
+    constructor(group, inKey) {
+        super(group, inKey);
+    }
+}
+
+class PushButton extends Button {
+    constructor(group, inKey) {
+        super(group, inKey);
+    }
+    input(pressed) {
+        engine.setValue(this.group, this.inKey, pressed);
+    }
+}
+
+class ToggleButton extends Button {
+    constructor(group, inKey) {
+        super(group, inKey);
+    }
+    input(pressed) {
+        if (pressed) {
+            script.toggleControl(this.group, this.inKey);
+        }
+    }
+}
+
+class Pot extends Component {
+    constructor(group, inKey, max) {
+        super(group, inKey);
+        this.max = max;
+        this.firstValueReceived = false;
+    }
+    input(value) {
+        engine.setParameter(this.group, this.inKey, value/this.max);
+        if (!this.firstValueReceived) {
+            this.firstValueReceived = true;
+            this.connect();
+        }
+    }
+    connect() {
+        engine.softTakeover(this.group, this.inKey, true);
+    }
+    disconnect() {
+        engine.softTakeoverIgnoreNextValue(this.group, this.inKey);
+    }
+}
+
+const getBit = (byte, index) => {
+    return (byte >> 8 - index) & 1;
```

Comment:
```suggestion
    return (byte >> (8 - index)) & 1;
```
Makes it a bit easier to read imo since its not clear what precedence the bitwise operators have...